File: sol/2419/2419.cpp
struct Solution {
    int longestSubarray(vector<int>& nums) {
        int n = nums.size(), maxele = nums[0], maxlen = 1;
        int currele = maxele, currlen = 1;
        for (int i = 1; i < n; ++i){
            if (nums[i] == currele){
                ++currlen;
            } else {
                currele = nums[i];
                currlen = 1;
            }
            if (currele > maxele || (currele == maxele && currlen > maxlen)){
                maxele = currele;
                maxlen = currlen;
            }
        }
        return maxlen;
    }
};

File: sol/2419/2419.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        maxele = nums[0]
        maxlen = 1
        currele = nums[0]
        currlen = 1
        for i in range(1, n):
            if nums[i] == currele:
                currlen += 1
            else:
                currele = nums[i]
                currlen = 1
            if currele > maxele or (currele == maxele and currlen > maxlen):
                maxele = currele
                maxlen = currlen
        return maxlen

